packages/world-modules/src/utils/SystemSwitch.sol

```diff
@@ -39,19 +39,27 @@ library SystemSwitch {
   function call(ResourceId systemId, bytes memory callData) internal returns (bytes memory returnData) {
     address worldAddress = WorldContextConsumerLib._world();
 
-    // If we're in the World context, call the system directly via delegatecall
+    // If we're in the World context
     if (address(this) == worldAddress) {
       (address systemAddress, ) = Systems.get(systemId);
```

Comment:
looking at this I wonder if we should just use the `SystemCall.call` library here


```solidity
  function call(ResourceId systemId, bytes memory callData) internal returns (bytes memory returnData) {
    address worldAddress = WorldContextConsumerLib._world();

    // If we're in the World context, call via the internal library
    if (address(this) == worldAddress) {
      return SystemCall.call(systemId, callData);
    }

    // Otherwise, call the system via world.call
    returnData = IWorldKernel(worldAddress).call(systemId, callData);
  }
```

Reply:
I like this! It adds a small amount of gas for the `publicAccess` and `value` checks but I like that we can reuse the code.

Reply:
should we change all the other places to use the same lib?

Reply:
Are there other places that do this? Searching for `delegatecallWithContext` or `== ROOT_NAMESPACE` doesn't bring up much for me

Reply:
I wonder if this line could cause an error because it updates the balances expecting a root

`if (value > 0) {`